lib/Doctrine/ORM/Mapping/DiscriminatorColumnMapping.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+use BackedEnum;
+
+use function in_array;
+use function property_exists;
+
+/**
+ * @template-implements ArrayAccess<string, mixed>
+ * @psalm-type DiscriminatorColumnMappingType = 'boolean'|'array'|'object'|'datetime'|'time'|'date'
+ */
+final class DiscriminatorColumnMapping implements ArrayAccess
+{
+    use ArrayAccessImplementation;
+
+    /** @var int|null The database length of the column. Optional. Default value taken from the type. */
+    public int|null $length = null;
+
+    public string|null $columnDefinition = null;
+
+    /** @var class-string<BackedEnum>|null */
+    public string|null $enumType = null;
+
+    /** @param DiscriminatorColumnMappingType $type */
+    public function __construct(
+        public string $type,
+        public string $fieldName,
+        public string $name,
+    ) {
+    }
+
+    /** @param array{type: DiscriminatorColumnMappingType, fieldName: string, name: string} $mappingArray */
+    public static function fromMappingArray(array $mappingArray): self
+    {
+        $mapping = new self(
+            $mappingArray['type'],
+            $mappingArray['fieldName'],
+            $mappingArray['name'],
+        );
+        foreach ($mappingArray as $key => $value) {
+            if (in_array($key, ['type', 'fieldName', 'name'])) {
+                continue;
+            }
+
+            if (property_exists($mapping, $key)) {
+                $mapping->$key = $value;
+            }
```

Comment:
In other classes, you throw for invalid properties while you silently ignore them here. Is that on purpose?

Reply:
IIRC I started added that check in other classes after this one, I must have forgotten to update it.